src/planscape/forsys/parse_forsys_output_test.py

```diff
@@ -509,3 +510,133 @@ def _get_raw_forsys_output(self) -> ro.vectors.ListVector:
+        with self.assertRaises(Exception) as context:
+            # area header is "area_ha" instead of "area"
+            ForsysGenerationOutputForASingleScenario(
+                raw_forsys_output, {"p1": 1, "p2": 2},
+                "proj_id", "area_ha", "cost", "geo_wkt")
+
+        self.assertEqual(
+            str(context.exception),
+            'header, ETrt_area_ha, is not a forsys output header')
+
+    def test_fails_if_cost_header_is_wrong(self) -> None:
+        raw_forsys_output = self._get_raw_forsys_output()
+
+        with self.assertRaises(Exception) as context:
+            # cost header is "c" instead of "cost"
+            ForsysGenerationOutputForASingleScenario(
+                raw_forsys_output, {"p1": 1, "p2": 2},
+                "proj_id", "area", "c", "geo_wkt")
+
+        self.assertEqual(
+            str(context.exception),
+            'header, ETrt_c, is not a forsys output header')
+
+    def test_fails_if_geo_wkt_header_is_wrong(self) -> None:
+        raw_forsys_output = self._get_raw_forsys_output()
+
+        with self.assertRaises(Exception) as context:
+            # geo_wkt header is "geometry" instead of "geo_wkt"
+            ForsysGenerationOutputForASingleScenario(
+                raw_forsys_output, {"p1": 1, "p2": 2},
+                "proj_id", "area", "cost", "geometry")
+
+        self.assertEqual(
+            str(context.exception),
+            'header, geometry, is not a forsys output header')
+
+    def _get_raw_forsys_output(self) -> ro.vectors.ListVector:
+        stand_output = {
+            "stand_id": [5, 7, 9, 1],
+            "proj_id": [2, 2, 1, 3],
+            "geo_wkt": [
+                MergePolygonsTest._create_polygon(
+                    self,
+                    ((0, 0), (0, 1), (1, 1),
+                     (1, 0), (0, 0))).wkt,
+                MergePolygonsTest._create_polygon(
+                    self,
+                    ((0, 0), (0, -1), (1, -1),
+                     (1, 0), (0, 0))).wkt,
+                MergePolygonsTest._create_polygon(
+                    self,
+                    ((0, 0), (0, -1), (1, -1),
+                     (1, 0), (0, 0))).wkt,
+                MergePolygonsTest._create_polygon(
+                    self, ((0, 0), (0, -1), (1, -1),
+                           (1, 0), (0, 0))).wkt]}
+        project_output = {
+            "proj_id": [2, 1, 3],
+            "ETrt_p1": [0.1, 0.5, 0.3],
+            "ETrt_p2": [0.4, 0.1, 0.1],
+            "treatment_rank": [1, 2, 3],
+            "ETrt_area": [11, 10, 12],
+            "ETrt_cost": [600, 500, 800],
+        }
+        raw_forsys_output = ro.vectors.ListVector(
+            {"stand_output": _convert_dictionary_of_lists_to_rdf(
+                self,
+                stand_output),
+             "project_output": _convert_dictionary_of_lists_to_rdf(
+                self,
+                project_output),
+             "subset_output": _convert_dictionary_of_lists_to_rdf(self, {})})
```

Comment:
out of curiosity, what is a subset_output and do we use it?

Reply:
I'm unsure: I haven't played with subset_output at all.
So I'm a little curious, too.

Looking at the test_forest ranking example, subset_output doesn't seem particularly different from project_output.
![image](https://user-images.githubusercontent.com/3720361/221695376-5bebf9ba-a5d1-463b-a4e6-c1181597fe70.png)